.github/workflows/build-indy-node-monitor.yaml

```diff
@@ -0,0 +1,77 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'fetch-validator-status/**'  # Build only if their are changes in this directory
+  
+  workflow_dispatch:
+    inputs:
+      ref:
+        required: false
+        type: string
+        description: "This is the tag version, don't include the v like this 2.4.6, if you dont provide a version we will get the latest tag"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository main
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Required due to the way Git works, without it this action won't be able to find any or the correct tags
+
+      - name: Extract latest version from Git tag, excludind the v.
+        run: |
+          LATEST_TAG=$(git describe --tags `git rev-list --tags --max-count=1` | cut -c2-)
+          echo "LATEST_TAG=$LATEST_TAG" >> $GITHUB_ENV
+          echo "LATEST_TAG=$LATEST_TAG"
+
+      - name: Checkout repository specific tag
+        uses: actions/checkout@v4
+        with:
+          ref: v${{ inputs.ref || env.LATEST_TAG }}
+
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set lowercase username
+        run: echo "USERNAME=$(echo ${{ github.repository_owner }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+      - name: Checkout repository main
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Required due to the way Git works, without it this action won't be able to find any or the correct tags
+
+      - name: Extract latest version from Git tag, excludind the v.
+        run: |
+          LATEST_TAG=$(git describe --tags `git rev-list --tags --max-count=1` | cut -c2-)
+          echo "LATEST_TAG=$LATEST_TAG" >> $GITHUB_ENV
+          echo "LATEST_TAG=$LATEST_TAG"
+
+      - name: Checkout repository specific tag
+        uses: actions/checkout@v4
+        with:
+          ref: v${{ inputs.ref || env.LATEST_TAG }}
+
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set lowercase username
+        run: echo "USERNAME=$(echo ${{ github.repository_owner }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
```

Comment:
This whole section of code seems to be duplicated.

Reply:
Yes if we go the TAG way we first need to checkout de main code and then get the latest TAG from the code and then we can chechout the correct version for the build.

Since we limit the change to the fetch-valdator-status path the build will not be launch very often but going with the release make sense.

Right now I am asinged on a new task in Sylvain team, but if you want to go the relaese way that would be a good way forward, but I will have limited bandwith for the next few weeks.

Best.